ivadomed/utils.py

```diff
@@ -235,14 +239,23 @@ def structurewise_uncertainty(fname_lst, fname_hard, fname_unc_vox, fname_out):
         nib_im = nib.load(fname)
         data_im = nib_im.get_fdata()
         data_lst.append(data_im)
-        data_im_l, _ = label(data_im, structure=bin_struct)
+        data_im_l = []
+        for i in range(data_im.shape[3]):
+            data_im_tmp, _ = label(data_im[:, :, :, i], structure=bin_struct)
+            data_im_l.append(data_im_tmp)
         data_l_lst.append(data_im_l)
         del nib_im
+    data_l_lst = np.array(data_l_lst)
+    # channel went first due to the 'append' function
+    data_l_lst = np.reshape(data_l_lst, (data_l_lst.shape[0], data_l_lst.shape[2], data_l_lst.shape[3],
+                                         data_l_lst.shape[4], data_l_lst.shape[1]))
```

Comment:
I tend to use "transpose" for these kind of things, eg [here](https://github.com/ivadomed/ivadomed/blob/master/ivadomed/testing.py#L225).
Is it doing the same thing?

Reply:
I think so. I am just permuting two axis so yes.

Reply:
done in 946dd6a